src/group_theory/sylow.lean

```diff
@@ -21,17 +21,127 @@ In this file, currently only the first of these results is proven.
 
 ## Main statements
 
-* `exists_subgroup_card_pow_prime`: A generalisation of the first of the Sylow theorems: For every
-  prime power `pⁿ` dividing `G`, there exists a subgroup of `G` of order `pⁿ`.
+* `exists_subgroup_card_pow_prime`: A generalization of Sylow's first theorem:
+  For every prime power `pⁿ` dividing `G`, there exists a subgroup of `G` of order `pⁿ`.
+* `is_p_group.exists_le_sylow`: A generalization of Sylow's first theorem:
+  Every `p`-subgroup is contained in a Sylow `p`-subgroup.
+* `sylow_conjugate`: A generalization of Sylow's second theorem:
+  If the number of Sylow `p`-subgroups is finite, then all Sylow `p`-subgroups are conjugate.
+* `card_sylow_modeq_one`: A generalization of Sylow's third theorem:
+  If the number of Sylow `p`-subgroups is finite, then it is congruent to `1` modulo `p`.
+-/
 
-## TODO
+open fintype mul_action subgroup
 
-* Prove the second and third of the Sylow theorems.
-* Sylow theorems for infinite groups
--/
+section infinite_sylow
+
+variables (p : ℕ) (G : Type*) [group G]
+
+/-- A Sylow `p`-subgroup is a maximal `p`-subgroup. -/
+def sylow :=
+{P : subgroup G // is_p_group p P ∧ ∀ {Q : subgroup G}, is_p_group p Q → P ≤ Q → Q = P}
```

Comment:
I wonder if it would be easier to make this a structure that `extends subgroup` and has two extra fiels: `is_p_group` and `is_maximal`.

Reply:
Sure. I'm not familiar enough with structures to know what difference it will make, but the code seems to compile fine, so I'll push the change.